pkg/controllers/hub/trafficmanagerbackend/controller.go

```diff
@@ -197,15 +188,15 @@ func (r *Reconciler) cleanupEndpoints(ctx context.Context, backend *fleetnetv1be
 			continue // skipping deleting the endpoints which are not created by this backend
 		}
 		errs.Go(func() error {
-			if _, err := r.EndpointsClient.Delete(cctx, r.ResourceGroupName, atmProfileName, armtrafficmanager.EndpointTypeAzureEndpoints, *endpoint.Name, nil); err != nil {
+			if _, err := r.EndpointsClient.Delete(cctx, resourceGroup, atmProfileName, armtrafficmanager.EndpointTypeAzureEndpoints, *endpoint.Name, nil); err != nil {
 				if azureerrors.IsNotFound(err) {
-					klog.V(2).InfoS("Ignoring NotFound Azure Traffic Manager endpoint", "trafficManagerBackend", backendKObj, "atmProfileName", atmProfileName, "atmEndpoint", *endpoint.Name)
+					klog.V(2).InfoS("Ignoring NotFound Azure Traffic Manager endpoint", "resourceGroup", resourceGroup, "trafficManagerBackend", backendKObj, "atmProfileName", atmProfileName, "atmEndpoint", *endpoint.Name)
 					return nil
 				}
-				klog.ErrorS(err, "Failed to delete the endpoint", "trafficManagerBackend", backendKObj, "atmProfileName", atmProfileName, "atmEndpoint", *endpoint.Name)
+				klog.ErrorS(err, "Failed to delete the endpoint", "resourceGroup", resourceGroup, "trafficManagerBackend", backendKObj, "atmProfileName", atmProfileName, "atmEndpoint", *endpoint.Name)
```

Comment:
As per discussion, it would be good to emit events about the deletion failures so that cx can understand why deletion is stuck and fix.

Reply:
it will be in a separate PR to add events for all the atm controllers.